Source/CarthageKit/Project.swift

```diff
@@ -684,50 +698,85 @@ public final class Project {
 			.then(SignalProducer<(), CarthageError>.empty)
 	}
 
+	/// - parameters:
+	/// 	- flatMapTransform: defaults to identity function. Only ever used for one thing, but parameterized for unit testing purposes.
+	private func fileSystemObjects(for dependency: Dependency<PinnedVersion>, withRepository repositoryURL: URL, at pathList: String, flatMapTransform: @escaping (String) -> String? = { $0 }) -> SignalProducer<[String], CarthageError> {
```

Comment:
We'd need to collect them again to filter dependencies with. For (future-written) unit testing, we want to check against an array. Can't really think of an instance wqh